pkg/apis/capabilities/v1beta1/product_types.go

```diff
@@ -0,0 +1,1095 @@
+	if s.Deployment == nil {
+		return nil
+	}
+	deploymentOption := s.Deployment.DeploymentOption()
+	return &deploymentOption
+}
+
+func (s *ProductSpec) AuthenticationMode() *string {
+	if s.Deployment == nil {
+		return nil
+	}
+	return s.Deployment.AuthenticationMode()
+}
+
+func (s *ProductSpec) ProdPublicBaseURL() *string {
+	if s.Deployment == nil {
+		return nil
+	}
+	return s.Deployment.ProdPublicBaseURL()
+}
+
+func (s *ProductSpec) StagingPublicBaseURL() *string {
+	if s.Deployment == nil {
+		return nil
+	}
+	return s.Deployment.StagingPublicBaseURL()
+}
+
+func (s *ProductSpec) SecuritySecretToken() *string {
+	if s.Deployment == nil {
+		return nil
+	}
+	return s.Deployment.SecuritySecretToken()
+}
+
+func (s *ProductSpec) HostRewrite() *string {
+	if s.Deployment == nil {
+		return nil
+	}
+	return s.Deployment.HostRewrite()
+}
+
+func (s *ProductSpec) CredentialsLocation() *string {
+	if s.Deployment == nil {
+		return nil
+	}
+	return s.Deployment.CredentialsLocation()
+}
+
+func (s *ProductSpec) AuthUserKey() *string {
+	if s.Deployment == nil {
+		return nil
+	}
+	return s.Deployment.AuthUserKey()
+}
+
+func (s *ProductSpec) AuthAppID() *string {
+	if s.Deployment == nil {
+		return nil
+	}
+	return s.Deployment.AuthAppID()
+}
+
+func (s *ProductSpec) AuthAppKey() *string {
+	if s.Deployment == nil {
+		return nil
+	}
+	return s.Deployment.AuthAppKey()
+}
+
+// ProductStatus defines the observed state of Product
+// +k8s:openapi-gen=true
+type ProductStatus struct {
+	// +optional
+	ID *int64 `json:"productId,omitempty"`
+	// +optional
+	State *string `json:"state,omitempty"`
+
+	// ObservedGeneration reflects the generation of the most recently observed MachineSet.
+	// +optional
+	ObservedGeneration int64 `json:"observedGeneration,omitempty"`
+
+	// In the event that there is a terminal problem reconciling the
+	// product, both ErrorReason and ErrorMessage will be set. ErrorReason
+	// will be populated with a succinct value suitable for machine
+	// interpretation, while ErrorMessage will contain a more verbose
+	// string suitable for logging and human consumption.
+	//
+	// These fields should not be set for transitive errors that a
+	// controller faces that are expected to be fixed automatically over
+	// time (like service outages), but instead indicate that something is
+	// fundamentally wrong with the Product's spec or the configuration of
+	// the product controller, and that manual intervention is required. Examples
+	// of terminal errors would be invalid combinations of settings in the
+	// spec, values that are unsupported by the product controller, or the
+	// responsible product controller itself being critically misconfigured.
+	//
+	// Any transient errors that occur during the reconciliation of Products
+	// can be added as events to the Product object and/or logged in the
+	// controller's output.
+	// +optional
+	// TODO enum
+	ErrorReason *string `json:"errorReason,omitempty"`
```

Comment:
What's the reasoning of ErrReason and ErrorMessage being attributes of the status section and not conditions?
Seems ok to me but I'd like to know the reasoning so we can apply same criteria in other objects

Reply:
As the comment says, they are final errors that cannot be fixed retrying or anything like that. Not related to conditions and first level error field in Status section. 

I agree it could be redundant now. There exists a condition type specific for that.

Reply:
Removing fields